library/Mockery/Expectation.php

```diff
@@ -558,6 +558,28 @@ public function andReturnUsing(...$args)
         return $this;
     }
 
+    /**
+     * Sets up a closure to return the nth argument from the expected method call
+     *
+     * @param int $index
+     * @return self
+     */
+    public function andReturnArgument($index)
+    {
+        if (!is_int($index) || $index < 0) {
+            throw new \InvalidArgumentException("Invalid argument index supplied. Index must be a positive integer.");
+        }
+        $closure = function (...$args) use ($index) {
+            if (isset($args[$index])) {
```

Comment:
Probably want `array_key_exists` here, the argument could legitimately be `null`. Test case to cover would be good to

Reply:
Good catch, I'll update both